lib/sensu/server/process.rb

```diff
@@ -644,6 +644,50 @@ def setup_client_monitor
         end
       end
 
+      # Determine stale check results, those that have not executed in
+      # a specified amount of time (check TTL). This method iterates
+      # through the client registry and check results for checks with
+      # a defined TTL value (in seconds). If a check result has a
+      # defined TTL, the time since last check execution (in seconds)
+      # is calculated. If the time since last execution is equal to or
+      # greater than the check TTL, a warning check result is
+      # published with the appropriate check output.
+      def determine_stale_check_results
+        @logger.info("determining stale check results")
+        @redis.smembers("clients") do |clients|
+          clients.each do |client_name|
+            @redis.smembers("result:#{client_name}") do |checks|
+              checks.each do |check_name|
+                result_key = "#{client_name}:#{check_name}"
+                @redis.get("result:#{result_key}") do |result_json|
+                  unless result_json.nil?
+                    check = MultiJson.load(result_json)
+                    next unless check[:ttl] && check[:executed]
+                    time_since_last_execution = Time.now.to_i - check[:executed]
+                    if time_since_last_execution >= check[:ttl]
+                      check[:output] = "Last check execution was "
```

Comment:
Bonus of building on the stored result: check handler and custom attributes etc.